android/app/src/main/python/mmrelay/matrix_utils.py

```
@@ -0,0 +1,2322 @@
+        bot_user_id = matrix_section["bot_user_id"]
+
+        # Manual method does not support device_id - use auth system for E2EE
+        e2ee_device_id = None
+
+    # Get matrix rooms from config
+    if "matrix_rooms" not in config:
+        logger.error("Configuration is missing 'matrix_rooms' section")
+        logger.error(
+            "Please ensure your config.yaml includes matrix_rooms configuration"
+        )
+        raise ValueError("Missing required 'matrix_rooms' configuration")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)